samples/sampler/stories/default/ContextualPopupDecorator.js

```diff
@@ -0,0 +1,54 @@
+import {ContextualPopupDecorator} from '@enact/agate/ContextualPopupDecorator';
+import BodyText from '../../../../../moonstone/BodyText';
+import Button from '@enact/agate/Button';
+import ri from '@enact/ui/resolution';
+import React from 'react';
+import {storiesOf} from '@storybook/react';
+
+import {boolean, select, text} from '@enact/storybook-utils/addons/knobs';
+import {mergeComponentMetadata} from '@enact/storybook-utils';
+import {action} from '@enact/storybook-utils/addons/actions';
+
+const ContextualButton = ContextualPopupDecorator(Button);
+ContextualButton.displayName = 'ContextualButton';
+
+const Config = mergeComponentMetadata('ContextualPopupDecorator', Button, ContextualButton);
+
+// NOTE: Something about the HOC is inhibiting accessing its defaultProps, so we're adding them here
+// manually. This can (should) be revisited later to find out why and a solution.
+Config.defaultProps = {
+	direction: 'below center',
+	open: true,
+	showCloseButton: true,
+	spotlightRestrict: 'self-first'
+};
+
+const renderPopup = () => (
+	<div>{text('popup string', {groupId: 'Popup'}, 'Hello Contextual Popup')}</div>
+);
+
+storiesOf('Agate', module)
+	.add(
+		'ContextualPopupDecorator',
+		() => (
+			<div style={{textAlign: 'center', marginTop: ri.unit(99, 'rem')}}>
+				<ContextualButton
+					direction={select('direction', ['above', 'above center', 'above left', 'above right', 'below', 'below center', 'below left', 'below right', 'left middle', 'left top', 'left bottom', 'right middle', 'right top', 'right bottom'], Config)}
```

Comment:
I think that we need a better way to define the candidate prop values here from the component code. But for now, it's fine to me.